homework.js

```diff
@@ -0,0 +1,147 @@
+///////////////////////////////////////////////////////////////////////
+const countUpAndDown = function(num){
+    let result = ""
+    for(i = 0; i<num; i++){
+        result += (i+1)+" "
+    }
+    for(i = num-1; i>0; i--){
+        result += i+" "
+    }
+    return result 
+}
+console.log(countUpAndDown(10))
+///////////////////////////////////////////////////////////////////////
+const findLetterWithA = function (arr){
+    let result = []
+    for(i = 0; i<arr.length; i++){
+        for(j = 0; j<arr[i].length; j++){
+            if(arr[i][j] == "a"){
+                result.push(arr[i])
+                break
+            }
+                
+        }
+    }
+    return result
+}
+console.log(findLetterWithA(['cat', 'rabbit', 'dog', 'frog']))
+///////////////////////////////////////////////////////////////////////
+const findLetterWith = function (letter ,arr){
+    let result = []
+    for(i = 0; i<arr.length; i++){
+        for(j = 0; j<arr[i].length; j++){
+            if(arr[i][j] == letter){
+                result.push(arr[i])
+                break
+            }
+                
+        }
+    }
+    return result
+}
+console.log(findLetterWith("o",['cat', 'rabbit', 'dog', 'frog']))
+///////////////////////////////////////////////////////////////////////
+const longestWord = function(string){
+    let word = ""
+    let arr = string.split(" ");
+    for(i = 0; i<arr.length; i++){
+        if(arr[i].length > word.length){
+            word = arr[i]
+        }
+    }
+    return word
+}
+console.log(longestWord("The cat in the house"))
+///////////////////////////////////////////////////////////////////////
+const largestEvenNumber = function(arr){
+    let num = 0 
+    for(i = 0; i<arr.length; i++){
+        if(arr[i]%2==0){
+            if(arr[i] > num){
+                num = arr[i]
+           }
+        }   
+    }
+    return num
+}
+console.log(largestEvenNumber([1,2,3,10,40,17,0]))
+///////////////////////////////////////////////////////////////////////
+const wordLetters     = ['G', 'O', 'A', 'T'];
+const guessedLetters  = ['_', '_', '_', '_'];
+const guessLetter = function(letter){
+    let counter = 0
+    let c ;
+
+    for(i = 0; i<wordLetters.length; i++){
+        if(counter == guessedLetters.length-1){
+            return "You win"+ guessedLetters
+        }
+        
+        if(wordLetters[i] == letter){
+            guessedLetters[i] = letter
+            counter++
+            return "Correct "+ guessedLetters
+        }
+    }
+        return "Incorrect "+ guessedLetters
+}
+console.log(guessLetter('G')); // "Correct, G _ _ _"
+console.log(guessLetter('I')); // "Incorrect, G _ _ _"
+console.log(guessLetter('O')); // "Correct, G O _ _"
+console.log(guessLetter('A')); // "Correct, G O A _"
+console.log(guessLetter('L')); // "Incorrect, G O A _"
+console.log(guessLetter('T')); // "You Win, G O A T"
+///////////////////////////////////////////////////////////////////////
```

Comment:
I edited your code so the message `"You win"+ guessedLetters` show when the player win.
- I moved the variable `counter` outside of the function since we don't want it to get back to 0 each time the player guess a letter.
-  Changed `counter == guessedLetters.length-1` to `counter == wordLetters.length) as we want it to see if the number of the correct guess equal to the length of the word it self.
- changed the place of the if condition.

```
const wordLetters     = ['G', 'O', 'A', 'T'];
const guessedLetters  = ['_', '_', '_', '_'];
let counter = 0
const guessLetter = function(letter){

    let c ;

    for(let i = 0; i<wordLetters.length; i++){
 

        if(wordLetters[i] == letter){
            guessedLetters[i] = letter
            counter++
            if(counter == wordLetters.length){
              return "You win "+ guessedLetters
          }
            return "Correct "+ guessedLetters
        }
    }
        return "Incorrect "+ guessedLetters
}
console.log(guessLetter('G')); // "Correct, G _ _ _"
console.log(guessLetter('I')); // "Incorrect, G _ _ _"
console.log(guessLetter('O')); // "Correct, G O _ _"
console.log(guessLetter('A')); // "Correct, G O A _"
console.log(guessLetter('L')); // "Incorrect, G O A _"
console.log(guessLetter('T')); // "You Win, G O A T"
```